source/dofs/dof_tools_constraints.cc

```diff
@@ -2094,23 +2096,26 @@ namespace DoFTools
             }
 
           // In case of a dependency cycle we, either
-          //  - do nothing if cycle_factor == 1. In this case all degrees
+          //  - do nothing if cycle_constraint_factor == 1. In this case all
+          //  degrees
           //    of freedom are already periodically constrained,
           //  - otherwise, force all dofs to zero (by setting dof_left to
           //    zero). The reasoning behind this is the fact that
-          //    cycle_factor != 1 occurs in situations such as
+          //    cycle_constraint_factor != 1 occurs in situations such as
           //    x1 == x2 and x2 == -1. * x1. This system is only solved by
           //    x_1 = x_2 = 0.
 
           if (constraints_are_cyclic)
             {
-              if (std::abs(cycle_factor - 1.) > eps)
+              if (std::abs(cycle_constraint_factor / periodicity_factor - 1.) >
+                  eps)
```

Comment:
This is the solution that I took. It works, but I have the feeling that this could break in a corner case with an unstructured mesh. Is there a way to know if `dof_left` belongs to `face_1` and `dof_right` belongs to face_2?

Reply:
I do not understand how the mesh structure should play a role here. :-)

A bit above:
```
2045           auto dof_left  = dofs_1[j];                                            
2046           auto dof_right = dofs_2[i];
```
and the only place where this changes is
```
2055               std::swap(dof_left, dof_right);
```
So you could simply check whether `dof_left == dofs_1[j]`.

(I am about to hit the pillows, sorry for no coherent answer). The one thing I see that we have to check is the following: Let's say for a square domain with bloch periodicity on top and bottom and regular periodicity left and right, do we correctly identify the left and right point and set the phase shift between the upper and lower points? (without forcing the degree of freedom to 0.)